packages/dds/tree/src/simple-tree/schemaFactory.ts

```diff
@@ -567,11 +567,22 @@ export class SchemaFactory<
 	}
 
 	/**
-	 * Make a field of type identifier instead of the default, which is required.
-	 * @remarks Identifiers may be optionally supplied at node construction time.
-	 * If not supplied, they will be generated automatically when the node is inserted into the tree.
-	 * Attempting to read an automatically generated identifier before the node is inserted into the tree will throw an error.
-	 * An automatically generated identifier will not be present when iterating the nodes's fields until after the node is inserted into the tree.
+	 * A special field which holds a unique identifier for an object node.
+	 * @remarks
+	 * The value of this field, a "node identifier", uniquely identifies a node among all other nodes in the tree.
```

Comment:
On the one hand, if they don't provide an override ID, then it is uniquely identified among all nodes in _all trees_ since it is a UUID. On the other hand, if they provide an override (even one that is a UUID created automatically in the past, such as in the case of copy-pasting!), then there is no guarantee of uniqueness _at all_ and we should document that.
This makes me think that we should clarify here that there will, in the future, be a constraint that one can apply to insertions to ensure uniqueness and until then it is up to the application to guarantee uniqueness (though we provide UUIDs to help).

Reply:
Good thoughts, added some clarity.